mne/io/brainvision/tests/test_brainvision.py

```diff
@@ -469,4 +471,13 @@ def test_read_vmrk_annotations():
     except FileNotFoundError:
         pass
 
+    raw = read_raw_brainvision(vhdr_path)
+    events, event_id = events_from_annotations(raw)
+    for key, values in event_id.items():
+        assert int(key[-3:].lstrip()) == values
+
+    assert _bv_parser("Stimulus/S 11") == 11
+    assert _bv_parser("Stimulus/S202") == 202
+    assert _bv_parser("Response/R 12", add_for_r=100) == 112
```

Comment:
I don't like this much. You have a parameter that is not really exposed.
Also please use a regexp as this:

`int(re.split('(Stimulus|Response)/[RS]\s*', "Response/R 1")[-1])`

I am not it has the full logic but it's much more robust and easier to understand what pattern to match

Reply:
Should I just remove add_for_r from the call signature?

Reply:
Otherwise I changed it to this, is that what you mean:

```
def _bv_parser(description, add_for_r=1000):
    """Parse BrainVision event codes (like `Stimulus/S 11`) to ints."""
    try:
        *_, kind, code = re.split('(Stimulus|Response)/[RS]\s*', description)
        code = int(code)
        if kind == "Response":
            code += add_for_r
        return code
    except ValueError:  # probably not coercable to int, or no match at all
        warn("Marker '%s' could not be unambiguously mapped to an int and "
             "will be dropped." % description)
```

Reply:
can you use rather re.match rather than a try-catch?
but yes this for me really cleaner.

Reply:
@larsoner also see here